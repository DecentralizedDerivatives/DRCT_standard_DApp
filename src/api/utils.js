import axios from 'axios';

const extractData = response => {
  const milliseconds = 1000;
  return response.data.Data.map(obj => [obj.time * milliseconds, obj.close]);
};

export const get = async url => {
  let response, error;

  try {
    response = await axios.get(url);
  } catch (err) {
    error = err;
  }

  return response ? extractData(response) : error;
};
