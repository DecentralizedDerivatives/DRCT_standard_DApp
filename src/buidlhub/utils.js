export const propCheck = (props, reqd, ctx) => {
  if(!Array.isArray(reqd)) {
    throw new Error("Required props must be an array of required prop names");
  }
  if(!props || Object.keys(props).length < reqd.length) {
    throw new Error("Properties must have at least as many fields as required");
  }

  reqd.forEach(p=>{
    let v = props[p];

    if(typeof v === undefined || v === null) {
      let msg = ctx?`${ctx} is missing property: ${p}`:`Missing property: ${p}`;
      throw new Error(msg);
    }
  });
}
