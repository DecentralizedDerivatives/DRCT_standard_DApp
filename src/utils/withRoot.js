import React from 'react';
import {MuiThemeProvider} from 'material-ui/styles';
import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider';
import DateFnsUtils from 'material-ui-pickers/utils/date-fns-utils';
import theme from '../styles/muiTheme';

function withRoot(Component) {
  function WithRoot(props) {
    // MuiThemeProvider makes the theme available down the React tree
    // thanks to React context.
    return (
      <MuiThemeProvider theme={theme}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
          <Component {...props} />
        </MuiPickersUtilsProvider>
      </MuiThemeProvider>
    );
  }

  return WithRoot;
}

export default withRoot;
