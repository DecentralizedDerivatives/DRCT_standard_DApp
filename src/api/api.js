import {get} from './utils';

const btc = {
  get: () =>
    get(
      'https://min-api.cryptocompare.com/data/histohour?fsym=BTC&tsym=USD&limit=2000'
    ),
};

const eth = {
  get: () =>
    get(
      'https://min-api.cryptocompare.com/data/histohour?fsym=ETH&tsym=USD&limit=2000'
    ),
};

export default {
  eth,
  btc,
};
