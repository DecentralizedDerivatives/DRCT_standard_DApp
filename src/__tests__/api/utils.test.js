import mockAxios from 'jest-mock-axios';

import * as utils from '../../api/utils';

afterEach(() => {
  // cleaning up the mess left behind the previous test
  mockAxios.reset();
});
// jest.setTimeout(20000)
it('should call BTC url', () => {
  const data = utils.get('foo');
  expect(mockAxios.get).toHaveBeenCalledWith('foo');
});
// it('should raise error for fake url', async () => {
//   const data = await utils.get('foo')
//   console.log('DATA', data)
//   done()
// });
