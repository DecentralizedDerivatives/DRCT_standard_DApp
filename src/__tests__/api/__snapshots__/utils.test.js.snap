// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call url 1`] = `
Array [
  Array [
    0,
    10,
  ],
  Array [
    1000,
    20,
  ],
  Array [
    2000,
    30,
  ],
]
`;

exports[`should raise error 1`] = `[Error: error]`;
